### ISSUE TITLE
Due date are always shown with hours (french language)

### ISSUE THREAD

**Issue by molivier-fcms** [stefan-niedermann/nextcloud-deck#1034]
**Describe the bug**
With french language in my android UI, cards with due date are always shown with the date **and** the hour if the deadline is over 24h. If Android language is set in English, there is only the due date. It eats a precious space for cards with long title.

**Steps to reproduce the behavior**
1. (In android) go to `Settings` >  `General management` > `Language`
2. Add french language (displayed as `Français (France)`)
3. (In deck Android) Create a card with due date
4. The due date mentions the date and the hour whatever the date is in the future

**Expected behavior**
If due date is over 24h, the card should show only the date (without the hour).

**Screenshots**
French UI
![Screenshot_20210627-173734_Nextcloud Deck](https://user-images.githubusercontent.com/39309715/123551171-f44e1900-d770-11eb-823e-9262f555e2b7.jpg)

English UI
![Screenshot_20210627-174345_Nextcloud Deck](https://user-images.githubusercontent.com/39309715/123551173-f617dc80-d770-11eb-8ab9-7e9a47e500b8.jpg)

**Versions**
 - Nextcloud: 21.0.2
 - Nextcloud Deck: 1.4.2
 - Nextcloud Android: 3.16.1
 - Nextcloud Android Deck: 1.18.3

**Smartphone (please complete the following information):**
- Device: Sangung Galaxy A40
- Android-Version: 11
- App-Store:
  - [ ] Google Play Store
  - [ ] Google Play Store (Beta channel)
  - [x] F-Droid
  - [ ] Huawei AppGallery

**Comment by stefan-niedermann** [comment]
Indeed, i can confirm that behavior! Very strange, german and english behaves as expected. I'll have a look at this issue, thanks for the report 🙂.

**Comment by stefan-niedermann** [comment]
Now it gets even stranger. While i can reproduce the issue on my physical device, i don't see the problem on emulated devices (Android 5 and Android 12):

![grafik](https://user-images.githubusercontent.com/4741199/123644337-929bb680-d825-11eb-9670-86e0ffa3be19.png)

**Comment by stefan-niedermann** [comment]
Update: I can reproduce this issue on an emulated Android 5 device:
![grafik](https://user-images.githubusercontent.com/4741199/123644620-d55d8e80-d825-11eb-88f2-382904ace00b.png)

So maybe this isn't a bug in the Deck app at all but in Android itself (which might have been fixed in recent Android versions).

**Comment by stefan-niedermann** [comment]
Indeed: `DateUtils#getRelativeDateTimeString` returns on Android 5

```
10 avr. à 00:00
```
while it returns on Android 12

```
10 avr., 00:00
```

:thinking: hmm... looks like we need to rewrite this code part a bit... we are making assumptions about the returned format:

```java
if (parts.length == DATE_TIME_PARTS_SIZE) {
    if (parts[1].contains(":") && !parts[0].contains(":")) {
        return parts[0];
    } else if (parts[0].contains(":") && !parts[1].contains(":")) {
        return parts[1];
    }
}
```

**Comment by stefan-niedermann** [comment]
@molivier-fcms while a general clean solution will be quite comicated, i think i have at least found a solution for french language.

Would you mind testing [this debug build](https://github.com/stefan-niedermann/nextcloud-deck/suites/3162008149/artifacts/72760866)? You can install and use it safely parallel to the released version.

**Comment by stefan-niedermann** [comment]
The solution currently only eleminates `00:00`, i will try to enhance it so it behaves equally to english language. @desperateCoder can you check out the corresponding PR and replace the `.endsWith` part with an regex that goes for `2 digits - colon - 2 digits`?

**Comment by desperateCoder** [comment]
> can you check out the corresponding PR and replace the .endsWith part with an regex that goes for 2 digits - colon - 2 digits?

Sure, tomorrow at lunch break.

**Comment by molivier-fcms** [comment]
> @molivier-fcms while a general clean solution will be quite complicated, i think i have at least found a solution for french language.
> 
> Would you mind testing [this debug build](https://github.com/stefan-niedermann/nextcloud-deck/suites/3162008149/artifacts/72760866)? You can install and use it safely parallel to the released version.

Currently I still have the hour with the date with this debug build.

**Comment by desperateCoder** [comment]
Is it "00:00" again or maybe a different time? Could you please share a screenshot?

**Comment by molivier-fcms** [comment]
I've got this:
![1080-476-max](https://user-images.githubusercontent.com/39309715/124567700-8971a180-de44-11eb-82ff-0917fc9f6629.jpg)

The build version is 1.19.0, right?

**Comment by stefan-niedermann** [comment]
@molivier-fcms could you please test [this build](https://github.com/stefan-niedermann/nextcloud-deck/suites/3212663558/artifacts/74325657) again?

**Comment by stefan-niedermann** [comment]
ping @molivier-fcms [⇧⇧⇧](https://github.com/stefan-niedermann/nextcloud-deck/issues/1034#issuecomment-877987330)

**Comment by molivier-fcms** [comment]
Sorry I missed the first message. The new build works as expected.
I don't have the target hour anymore, which is good. Here the behavior I observe: 

- If event if below 1 hour: I see the remaining minutes
- If event is below 24 hours: I see remaining hours
- If event is below 48 hours: I see "tomorrow" (written in french)
- If event is below 72 hours: I see "After tomorrow" (written in french)
- If event is over 72 hours: I see "In X days" (written in french)

**Comment by stefan-niedermann** [comment]
Awesome, thanks for testing again!

We'll push this fix with `1.19.1`.
